### DISCUSSION TITLE
Ratio of inputs to outputs

### DISCUSSION THREAD

**Discussion by PatWalters** [emdgroup/baybe#431] [Q&A]
This is more of a philosophical question. I have a dataset with 5 inputs and 8 outputs (mostly quantities of impurities).  I don't have much experience with multi-objective GP.  Is there a rule of thumb for the ratio of the number of inputs to the number of outputs?

**Comment by AdrianSosic** [comment]
Hi @PatWalters, please apologize the late answer. Little time for philosophy at the moment 😄 I haven't seen any such rule so far and I doubt that one exists because it's probably impossible to quantify this in a general way. To me, this is a bit like the question "how many data points do I need to train a machine learning model" – it depends entirely on the context. Also, I think it depends on the specific aspect being asked and the structure of the model. For example, if your `N -> M` multi-objective GP (with `N` being input dimensions and `M` being output dimension) really is just a collection of `M` single-objective GPs of type `N -> 1`, then for each individual fit the number of output dimensions is completely irrelevant. In this case, the question is probably rather "for large `M`, can these individual GPs still lead to a coherent multi-objective decision"?

So my modest answer is: I honestly don't know 😬 Perhaps there are some theoretical results on this topic but at least I haven't seen anything like that.

**Comment by PatWalters** [comment]
@AdrianSosic, thanks for sharing your experience and releasing such a great package.
